instance/models/mixins/openedx_monitoring.py

```diff
@@ -42,28 +42,34 @@ def enable_monitoring(self):
             self.logger.warning('Skipping monitoring setup, '
                                 'NEWRELIC_ADMIN_USER_API_KEY not set')
             return
-        self.logger.info('Setting up New Relic Synthetics monitors')
-
-        # Delete existing monitors if they don't monitor this instance's
-        # public urls
-        monitors = newrelic.get_synthetics_monitors()
-        already_enabled = [monitor for monitor in monitors
-                           if monitor['uri'] in self._urls_to_monitor]
-        already_enabled_ids = {enabled['id'] for enabled in already_enabled}
-        for monitor in self.new_relic_availability_monitors.exclude(
-                pk__in=already_enabled_ids):
-            monitor.delete()
+        self.logger.info('Checking New Relic Synthetics monitors')
+
+        already_monitored_urls = set()
+
+        for monitor in self.new_relic_availability_monitors.all():
+            url = newrelic.get_synthetics_monitor(monitor.pk)['uri']
+            if url in self._urls_to_monitor:
+                already_monitored_urls.add(url)
+            else:
+                self.logger.info('Deleting New Relic Synthetics monitor for old public URL %s', url)
+                monitor.delete()
 
-        # Add monitors for urls that are not already being monitored
-        already_enabled_urls = {enabled['uri'] for enabled in already_enabled}
-        for url in self._urls_to_monitor - already_enabled_urls:
-            monitor_id = newrelic.create_synthetics_monitor(url)
-            self.new_relic_availability_monitors.create(pk=monitor_id)
+        for url in self._urls_to_monitor - already_monitored_urls:
+            self.logger.info('Creating New Relic Synthetics monitor for new public URL %s', url)
+            new_monitor_id = newrelic.create_synthetics_monitor(url)
+            self.new_relic_availability_monitors.create(pk=new_monitor_id)
 
-            # Set up email alerts
-            if settings.ADMINS:
-                emails = [email for name, email in settings.ADMINS]
-                newrelic.add_synthetics_email_alerts(monitor_id, emails)
+        # Set up email alerts.
+        # We add emails here but never remove them - that must be done manually (or the monitor deleted)
+        # in order to reduce the chance of bugs or misconfigurations accidentally supressing monitors.
+        emails_to_monitor = set([email for name, email in settings.ADMINS] + self.additional_monitoring_emails)
+        if emails_to_monitor:
+            for monitor in self.new_relic_availability_monitors.all():
+                emails_current = set(newrelic.get_synthetics_notification_emails(monitor.id))
+                emails_to_add = list(emails_to_monitor - emails_current)
+                if emails_to_add:
+                    self.logger.info('Adding email(s) to monitor %s: %s', monitor.id, ', '.join(emails_to_add))
+                    newrelic.add_synthetics_email_alerts(monitor.id, emails_to_add)
```

Comment:
Yes, I intentionally did not implement any way to automatically remove email addresses from monitors - see the comment block a few lines above (62-64). And yes, in part, that would prevent the unwanted removal of emails that were added via the web interface, even though that's not an encouraged way to configure monitors.

I think it will be pretty rare that we'll remove an email address from a monitor, but keep the monitor online. We can revisit this in the future if we find that that would be helpful, but for now I believe this is a reasonable and safe way forward.